src/Controls/src/Core/BindableObject.cs

```diff
@@ -150,8 +154,11 @@ void ClearValueCore(BindableProperty property, SetterSpecificity specificity)
 
 			if (changed)
 			{
-				OnPropertyChanged(property.PropertyName);
-				property.PropertyChanged?.Invoke(this, original.Value, newValue);
+				OnPropertyChangedBindableProperty(property, original.Value, newValue);
+			}
+			else
+			{
+				ProposeUpdateHandlerValue(property.PropertyName, false);
```

Comment:
The more I look at this the more I think we should try keep "handler" things out of BO for now.

Maybe this just needs to be a method that Element can override:

```cs
OnNotifyPropertyChanged(property, original.Value, newValue, changed);
```

and then in BO:

```cs
private protected virtual void OnPropertyChangedBindableProperty(BindableProperty property, object original, object value, bool changed)
{
    OnPropertyChanged(property.PropertyName);
	property.PropertyChanged?.Invoke(this, original.Value, newValue);
}
```

Then in Element, we can do:

```cs
private protected override void OnPropertyChangedBindableProperty(BindableProperty property, object original, object value, bool changed)
{
    if (changed)
    {
        _properties[property.PropertyName] = true;
        base.OnPropertyChangedBindableProperty(property, original, value, changed);
        _properties.Remove(property.PropertyName);
    }

    Handler?.UpdateValue(propertyName);
}
```

Not sure if that last logic is right, but since we make the same decision in BO to only trigger the handler if the property changes, we can rather put that logic into element and not have to do weird casting and things.

I also see the OnPropertyChnaged method in Element triggers the handler, so maybe I got it wrong...